front/app/modules/commercial/id_vienna_saml/components/ViennaSamlButton.tsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import { AUTH_PATH } from 'containers/App/constants';
+
+// i18n
+import messages from './messages';
+import { FormattedMessage } from 'utils/cl-intl';
+
+import { StyledAuthProviderButton } from 'components/SignUpIn/AuthProviders';
+import { TOnContinueFunction } from 'components/SignUpIn/AuthProviderButton';
+
+// typings
+import { TSignUpInFlow } from 'components/SignUpIn';
+
+interface Props {
+  flow: TSignUpInFlow;
+  onContinue: TOnContinueFunction;
+}
+
+const ViennaSamlButton = ({ onContinue, flow }: Props) => {
+  const setHref = () => {
```

Comment:
Why do you need to define this `setHref` function? Can it be the case that our default `setHref` is enough? https://github.com/CitizenLabDotCo/citizenlab/blob/2da6a39d67904e72451f49205a460806da543057/front/app/services/singleSignOn.ts#L36

Reply:
This is just reverting what already existed in the code base, I'd rather not touch it now.

Reply:
I think back then we did not use the existing method because it adds tracking parameters to the request, which made the SSO flow fail on Vienna's side.

Reply:
I mean calling `onContinue` without the second parameter

```ts
    } else {
      onContinue('id_vienna_saml', setHref);
    }
 ```

Reply:
> This is just reverting what already existed in the code base

Ah I didn't realize it. Good for me then.

Reply:
I don't know, @brentguf introduced the change here originally: https://github.com/CitizenLabDotCo/citizenlab/commit/949cd938ccca252ede93bf05338fc7ba2fdf2809